okhttp/src/main/java/okhttp3/internal/connection/StreamAllocation.java

```diff
@@ -106,10 +106,16 @@ public HttpCodec newStream(
     boolean connectionRetryEnabled = client.retryOnConnectionFailure();
 
     try {
+      RealConnection existingConnection = connection;
+
       RealConnection resultConnection = findHealthyConnection(connectTimeout, readTimeout,
           writeTimeout, connectionRetryEnabled, doExtensiveHealthChecks);
       HttpCodec resultCodec = resultConnection.newCodec(client, chain, this);
 
+      if (existingConnection != connection) {
```

Comment:
For tracing, it's awkward because I could use the connectionFound as a connectionStart/End with zero cost.  So I will just need to filter out the connectionFound when it's the same host.

If you think about a zipkin trace, I want a zero cost dot when its using an existing connection.  https://user-images.githubusercontent.com/231923/28271719-373be6b2-6abe-11e7-921c-c15bde40e127.png